chainer/training/triggers/interval.py

```diff
@@ -40,9 +42,16 @@ def __call__(self, trainer):
         """
         updater = trainer.updater
         if self.unit == 'epoch':
-            prev = self.count
-            self.count = updater.epoch_detail // self.period
-            return prev != self.count
+            epoch_detail = updater.epoch_detail
+            previous_epoch_detail = updater.previous_epoch_detail
```

Comment:
> So will you change the arguments of trigger.__call__()? Currently, it takes a trainer. I think this is specific.

No, we do not think of it. I am not sure I understand your question but what I wanted to say here is that developers of `trigger`s must assume that they are called every iteration, that restricts the usage of `trigger`s.